aten/cmake/TestForCudaHalf.cmake

```diff
@@ -0,0 +1,7 @@
+if(CUDA_HAS_FP16 OR NOT ${CUDA_VERSION} LESS 7.5)
```

Comment:
if we enforce CUDA >= 8.0, we can remove this file entirely

Reply:
I'll do this one in a separate PR, since I'll need to eliminate the macro while I'm at it.